magma/circuit.py

```diff
@@ -607,7 +607,8 @@ def __new__(metacls, name, bases, dct):
                                 "definition syntax is not supported, ignoring "
                                 "'definition'")
         run_unconnected_check = run_unconnected_check or (
-            has_definition and self._syntax_style_ is _SyntaxStyle.NEW)
+            has_definition and self._syntax_style_ is _SyntaxStyle.NEW) and \
+            not dct.get("IGNORE_UNDRIVEN", False)
```

Comment:
still not a fan of having this be a class-level flag, but not the worst thing if we really want to enable/disable this at a class granularity. If we do keep it it should be `_ignore_undriven_` per Caleb's standard.

And would have this be on a separate line: `run_unconnected_check  = run_unconnected_check and not dct.get("_ignore_undriven_", False)`

Reply:
Would you prefer it to be a a global flag? I think the rational is that the verification module wants to ignore these errors, but we should keep it on for the RTL so we want to guard it per-circuit.

Reply:
ok, seems like there is a need for it per-circuit. fine with this design